<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="currentIndex">
       <number>3</number>
      </property>
      <widget class="QWidget" name="tab1">
       <attribute name="title">
        <string>Generate KeyPair</string>
       </attribute>
       <widget class="QTextEdit" name="Name">
        <property name="geometry">
         <rect>
          <x>11</x>
          <y>34</y>
          <width>451</width>
          <height>41</height>
         </rect>
        </property>
       </widget>
       <widget class="QTextEdit" name="Email">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>120</y>
          <width>451</width>
          <height>41</height>
         </rect>
        </property>
       </widget>
       <widget class="QRadioButton" name="RSA">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>210</y>
          <width>51</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>RSA</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">AsymmetricAlgoGroup</string>
        </attribute>
       </widget>
       <widget class="QRadioButton" name="DSA">
        <property name="geometry">
         <rect>
          <x>80</x>
          <y>210</y>
          <width>51</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>DSA</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">AsymmetricAlgoGroup</string>
        </attribute>
       </widget>
       <widget class="QRadioButton" name="KeySize_1024">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>280</y>
          <width>55</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>1024</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">KeySizeGroup</string>
        </attribute>
       </widget>
       <widget class="QRadioButton" name="KeySize_2048">
        <property name="geometry">
         <rect>
          <x>90</x>
          <y>280</y>
          <width>55</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>2048</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">KeySizeGroup</string>
        </attribute>
       </widget>
       <widget class="QLabel" name="label">
        <property name="geometry">
         <rect>
          <x>11</x>
          <y>11</y>
          <width>33</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Name</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_2">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>90</y>
          <width>31</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Email</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_3">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>180</y>
          <width>185</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Asymmetric encription algorithm</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_4">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>250</y>
          <width>46</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Key size</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="ElGamal">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>210</y>
          <width>73</width>
          <height>20</height>
         </rect>
        </property>
        <property name="toolTipDuration">
         <number>2</number>
        </property>
        <property name="text">
         <string>ElGamal</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">AsymmetricAlgoGroup</string>
        </attribute>
       </widget>
       <widget class="QPushButton" name="GenerateButton">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>340</y>
          <width>121</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Generate</string>
        </property>
       </widget>
       <widget class="QLabel" name="ErrorMsg">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>390</y>
          <width>291</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab2">
       <attribute name="title">
        <string>Private Key Ring</string>
       </attribute>
       <widget class="QTableWidget" name="privateKeyRingTable">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>170</y>
          <width>751</width>
          <height>341</height>
         </rect>
        </property>
        <property name="rowCount">
         <number>0</number>
        </property>
       </widget>
       <widget class="QPushButton" name="importPrivateKeyButton">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>120</y>
          <width>131</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Import</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_5">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>10</y>
          <width>201</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Private key .pem path</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_6">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>60</y>
          <width>151</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Public key .pem path</string>
        </property>
       </widget>
       <widget class="QPlainTextEdit" name="privateKeyImportPath">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>30</y>
          <width>381</width>
          <height>31</height>
         </rect>
        </property>
       </widget>
       <widget class="QPlainTextEdit" name="publicKeyImportPath">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>80</y>
          <width>381</width>
          <height>31</height>
         </rect>
        </property>
       </widget>
       <widget class="QPlainTextEdit" name="passphraseImportField">
        <property name="geometry">
         <rect>
          <x>430</x>
          <y>30</y>
          <width>131</width>
          <height>31</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="label_7">
        <property name="geometry">
         <rect>
          <x>430</x>
          <y>0</y>
          <width>331</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Passphrase (used for both import and export)</string>
        </property>
       </widget>
       <widget class="QLabel" name="privateKeyRingError">
        <property name="geometry">
         <rect>
          <x>430</x>
          <y>70</y>
          <width>331</width>
          <height>51</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab3">
       <attribute name="title">
        <string>Public Key Ring</string>
       </attribute>
       <widget class="QPushButton" name="importPublicKeyButton">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>70</y>
          <width>131</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Import</string>
        </property>
       </widget>
       <widget class="QPlainTextEdit" name="importPathForPublicKey">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>381</width>
          <height>31</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="publicKeyRingError">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>120</y>
          <width>331</width>
          <height>51</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QTableWidget" name="publicKeyRingTable">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>120</y>
          <width>751</width>
          <height>391</height>
         </rect>
        </property>
        <property name="rowCount">
         <number>0</number>
        </property>
       </widget>
       <widget class="QLabel" name="label_10">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>151</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Public key .pem path</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab4">
       <attribute name="title">
        <string>Send Message</string>
       </attribute>
       <widget class="QPlainTextEdit" name="messageContentTextBox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>90</y>
          <width>751</width>
          <height>91</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="label_8">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>70</y>
          <width>111</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Message Content</string>
        </property>
       </widget>
       <widget class="QCheckBox" name="zipCheckbox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>290</y>
          <width>201</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Compress the message (ZIP)</string>
        </property>
       </widget>
       <widget class="QCheckBox" name="radix64Checkbox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>480</y>
          <width>261</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Convert data to radix-64 format</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="select3DES">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>360</y>
          <width>95</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>3DES</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">SymmetricAlgoGroup</string>
        </attribute>
       </widget>
       <widget class="QRadioButton" name="selectAES128">
        <property name="geometry">
         <rect>
          <x>80</x>
          <y>360</y>
          <width>95</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>AES128</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">SymmetricAlgoGroup</string>
        </attribute>
       </widget>
       <widget class="QLabel" name="label_9">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>340</y>
          <width>251</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Select symmetric encryption algorithm:</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_11">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>390</y>
          <width>351</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Recipients Public Key ID used for encrypting the session key</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="doNotEncrypt">
        <property name="geometry">
         <rect>
          <x>160</x>
          <y>360</y>
          <width>141</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Do not encrypt</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">SymmetricAlgoGroup</string>
        </attribute>
       </widget>
       <widget class="QLabel" name="label_12">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>200</y>
          <width>431</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Private Key ID used for signature (leave empty if you do not want to sign)</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_13">
        <property name="geometry">
         <rect>
          <x>450</x>
          <y>310</y>
          <width>271</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Enter the path where message will be saved:</string>
        </property>
       </widget>
       <widget class="QPlainTextEdit" name="pathTextBox">
        <property name="geometry">
         <rect>
          <x>450</x>
          <y>340</y>
          <width>261</width>
          <height>31</height>
         </rect>
        </property>
       </widget>
       <widget class="QPushButton" name="sendMessageButton">
        <property name="geometry">
         <rect>
          <x>510</x>
          <y>430</y>
          <width>131</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Send Message</string>
        </property>
       </widget>
       <widget class="QPlainTextEdit" name="privateKeyTextBox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>220</y>
          <width>271</width>
          <height>31</height>
         </rect>
        </property>
       </widget>
       <widget class="QPlainTextEdit" name="publicKeyTextbox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>410</y>
          <width>271</width>
          <height>31</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="label_14">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>111</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Filename</string>
        </property>
       </widget>
       <widget class="QPlainTextEdit" name="messageTitle">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>40</y>
          <width>261</width>
          <height>31</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="errorLabel">
        <property name="geometry">
         <rect>
          <x>450</x>
          <y>380</y>
          <width>281</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab5">
       <attribute name="title">
        <string>Receive Message</string>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionGenerate_new_keypair">
   <property name="text">
    <string>Generate new keypair</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="KeySizeGroup"/>
  <buttongroup name="AsymmetricAlgoGroup"/>
  <buttongroup name="SymmetricAlgoGroup"/>
 </buttongroups>
</ui>
